data_hub/views/api_views.py

```diff
@@ -40,3 +43,34 @@ def get(self, request):
             return JsonResponse(json.loads(df), safe=False)
 
         return JsonResponse(serializer.errors, status=400)
+
+
+class MacroeconomicAPIView(APIView):
+    def get(self, request):
+
+        serializer = MacroeconomicAPIParamSerializer(data=request.query_params)
+        data_service = DataService()
+        if serializer.is_valid():
```

Comment:
```suggestion
    def get(self, request):
        serializer = MacroeconomicAPIParamSerializer(data=request.query_params)
        
        if serializer.is_valid():
            data_service = DataService()
```

Initialization of `DataService` is expensive, so do it only when absolutely necessary (you're only using it inside this if statement, so keep it there)

nit: is better to not start code inside functions with a blank line :)